### PR TITLE
Issue #8328: Change name of checkstyle types to be close to real java types - regexp

### PR DESCRIPTION
**Pull request by gaurabdg**
#8328 

- Introuduced `Pattern` and `Pattern[]`
- Sorted the property types

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 1]
```diff
@@ -1011,10 +1011,23 @@ else if (fieldClass == URI.class) {
...
+                result = "Regular Expression";
+            }
+            else {
+                result = "Pattern";
+            }
         }
         else if (fieldClass == Pattern[].class) {
-            result = "Regular Expressions";
+            if ("ImportOrder:groups".equals(checkProperty)
+                || "ImportOrder:staticGroups".equals(checkProperty)) {
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:1025]: why this things are exceptional?

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 2]
```diff
@@ -1011,10 +1011,23 @@ else if (fieldClass == URI.class) {
             result = "URI";
         }
         else if (fieldClass == Pattern.class) {
-            result = "Regular Expression";
+            if ("SuppressionSingleFilter:checks".equals(checkProperty)
+                || "SuppressionXpathSingleFilter:files".equals(checkProperty)
+                || "IllegalTokenText:format".equals(checkProperty)) {
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:1016]: same here, why it is an exception/

File: src/xdocs/property_types.xml [hunk 3]
```diff
@@ -908,56 +927,52 @@
...
+              The token must be at the end of the line. For example:
               <div class="wrapper">
                 <pre>
-              &#47;**
-              * Summary text.
-              * More details.
-              *&#47;
-              public void method();
+    someVariable = aBigVariableNameToMakeThings + "this may work" +
+                   lookVeryInteresting;
```
Comment (strkkk) [inline src/xdocs/property_types.xml:971]: I would suggest to move these sorting changes to new PR to not mix it with renaming changes. Also, please add screenshot or list how new order looks like (to make sure it is sorted properly, it is not clear from diff)

File: src/xdocs/config_metrics.xml [hunk 4]
```diff
@@ -696,7 +696,7 @@ class Foo {
             <tr>
               <td>excludeClassesRegexps</td>
               <td>Specify user-configured regular expressions to ignore classes.</td>
-              <td><a href="property_types.html#regexp">Regular Expressions</a></td>
+              <td><a href="property_types.html#Pattern.5B.5D">Pattern[]</a></td>
```
Comment (strkkk) [inline src/xdocs/config_metrics.xml:699]: just wonder - shouldn't this be `%5B%5D` instead?

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java (hunk 2 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:1016]: while discussion is not finished on how to deal with this Checks, lets keep them as they were on  "Regular Expression" (not real java type), See details in telegram group.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java (hunk 1 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:1025]: while discussion is not finished on how to deal with this Checks, lets keep them as they were on  "Regular Expression" (not real java type), See details in telegram group.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:1025]: as we already have such type "Regular Expressions" in doc, we will keep it and even place in metadata for now, to let plugins know that type is very special and specified in words only in chekcstyle doc.

File: src/xdocs/config_metrics.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/config_metrics.xml:699]: https://github.com/checkstyle/checkstyle/blob/cf08aa01021f064a99e7816a280ba89143623442/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java#L152

it is `String[]` in setter, so  real type is `String[]`, if we want  to mean regexp nature of it keep it `Regular Expressions`.
Please do the same by all others.

PLEASE keep in this PR only updates where setter type is `Pattern`, all other cases keep as  `Regular Expression` or ` `Regular Expressions`.

Comment (gaurabdg) [reply src/xdocs/config_metrics.xml:699]: @romani 
That is what I tried to mention here in my comment https://github.com/checkstyle/checkstyle/issues/8328#issuecomment-652964231 that there are more than 5 such cases where setter type and field class is different.

File: src/xdocs/property_types.xml (hunk 3 above)
Comment (gaurabdg) [reply src/xdocs/property_types.xml:971]: Okay, will do it asap.

File: src/xdocs/config_metrics.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/config_metrics.xml:699]: lets skip all complicated cases for now, and merge what is simple.
in this PR please skip all such cases, ones telegram discussion finished, we will tackle them.

Comment (gaurabdg) [reply src/xdocs/config_metrics.xml]: Done

File: src/xdocs/property_types.xml (hunk 3 above)
Comment (gaurabdg) [reply src/xdocs/property_types.xml]: @strkkk Done: https://github.com/checkstyle/checkstyle/pull/8400

**Comment by romani** [comment]
@gaurabdg , please rebase to resolve conflict.

File: src/xdocs/config_filters.xml [hunk 5]
```diff
@@ -152,35 +152,35 @@
...
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
                   <td><code>null</code></td>
                   <td>3.5</td>
               </tr>
               <tr>
                   <td>idFormat</td>
                   <td>Specify check ID pattern to suppress.</td>
-                  <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                  <td><a href="property_types.html#Pattern">Pattern</a></td>
```
Comment (romani) [inline src/xdocs/config_filters.xml:183]: it is String, please keep it `Regular Expression`, if setter is not `Pattern` it should be abstract non-existing type "Regular Expression" that user cannot instantiate of.

Apply the same to all other.

File: src/xdocs/config_metrics.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/config_metrics.xml]: @gaurabdg , please answer question from @strkkk .

File: src/xdocs/config_filters.xml (hunk 5 above)
Comment (gaurabdg) [reply src/xdocs/config_filters.xml]: Done

File: src/xdocs/config_metrics.xml (hunk 4 above)
Comment (gaurabdg) [reply src/xdocs/config_metrics.xml]: > just wonder - shouldn't this be `%5B%5D` instead?

The generated site contains link in this format.

**Comment by strkkk** [comment]
@gaurabdg CI is failing, there is no new line char in new files. 
Please fix and change config in your IDE to automatically add new line at the end of files

**Pull request merged**